hubspot/plugin.py

```diff
@@ -454,57 +454,95 @@ class HubSpotSearchCondition:
         "LTE",
         "GT",
         "GTE",
+        "HAS_PROPERTY",
+        "NOT_HAS_PROPERTY",
     ]
     value: HubSpotPropertyValue
 
 
-@purpose("Search contacts.")
+@dataclass
+class AndGroup:
+    """A group of conditions that must ALL be true (AND logic). Maximum of 6 conditions"""
+
+    conditions: List[HubSpotSearchCondition]
+
+
+@dataclass
+class SearchQuery:
+    """
+    Top-level search query that combines multiple AND groups with OR logic.
+    Any ONE of the AndGroups must match for a result to be returned.
+
+    Limitations:
+    - Maximum of 5 or_groups.
```

Comment:
(Same with these)

Reply:
I've added additional validation.

The API will also raise an error with very similar error messages.